code/clue.js

```diff
@@ -1,59 +1,254 @@
+    mystery.killer = randomSelector(suspects);
+    // This will change the background color of the card and show picture and detail about the suspect.
+    document.getElementById('killerCard').style.background = mystery.killer.color;
+    document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+    document.getElementById('killerImage').src = mystery.killer.image;
+    document.getElementById('killerAge').innerHTML = `${mystery.killer.age} years old`;
+    document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+    document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+}
+
+// This function will be invoked when you click on the favouriteWeapon card. 
+const pickWeapon = () => {
+    // Randomly selects a favouriteWeapon from the favouriteWeapons-array and adds that to the mystery object.
+    mystery.weapon = randomSelector(weapons);
+    document.getElementById('weaponCard').style.background = mystery.weapon.color
+    document.getElementById('weaponName').innerHTML = mystery.weapon.name;
+    document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight} kg `;
+    document.getElementById('weaponImage').src = mystery.weapon.image;
+}
+
+// This function will be invoked when you click on the favouriteRoom card.
+const pickRoom = () => {
+    // Randomly selects a favouriteWeapon from the rooms-array and adds that to the mystery object
+    mystery.room = randomSelector(rooms);
+    document.getElementById('roomName').innerHTML = mystery.room.name;
+    document.getElementById('roomImage').src = mystery.room.image;
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+document.getElementById('killerCard').onclick = pickKiller;
+document.getElementById('weaponCard').onclick = pickWeapon;
+document.getElementById('roomCard').onclick = pickRoom;
+
+// Function that will be invoked when you click the reveal-button. It prints out a message with facts about the murderer.
+const revealMystery = () => {
+    if (mystery.killer == "" || mystery.favouriteWeapon == "" || mystery.room == "") {
+        document.getElementById('mysteryDisplay').innerHTML = `
+    The murder can 't be resolved yet.`;
+    } else {
+        document.getElementById('mysteryDisplay').innerHTML = `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName},
+         in the ${mystery.room.name} with a ${mystery.weapon.name}. `;
+        document.getElementById('image').src = mystery.killer.image;
+    }
+}
+
+//flips the cards on onclick
+document.querySelectorAll('.flip-card-container').forEach(item => {
+    item.onclick = () => {
+        item.classList.toggle("flip");
+    }
```

Comment:
COOL! 
I was just thinking about why the killer/weapon/room changes before the cards flip. But maybe that is related to where in the JS file you put this function. Now it for eg. picks a killer and then flips the card. Is there any way yo change that?